### PR TITLE
More comparison operators

### PR DESCRIPTION
**Pull request by marcauberer**
- Add more comparison operators: `<`, `>`, `<=`, `>=`
- Fix bug, which caused a crash when feeding the compiler with integer values in CompStmts

### CODE REVIEW

File: compiler/cpp/src/analyzer/analyzer.cpp [hunk 1]
```diff
@@ -90,6 +90,7 @@ void checkDataTypeCompatibilityCompStmt(CompStmtExprAST* compStmt, const json& d
         case ValueExprAST::NUMBER_EXPR:
             if (!jsonKeyValue.is_number_integer())
                 throw std::runtime_error(jsonKeyValue.dump() + " is not an integer");
+            return;
```
Comment (Splines) [inline compiler/cpp/src/analyzer/analyzer.cpp:93]: The good old switch/case break 😇😅

File: compiler/cpp/src/interpreter/interpreter.cpp [hunk 2]
```diff
@@ -114,39 +114,41 @@ bool evaluateCompStatement(CompStmtExprAST* compStmt, const json& data) {
...
+        auto leftValue = keyValue.get<std::string>();
         if (compStmt->getValue()->getType() == ValueExprAST::STRING_EXPR) {
-            auto *expectedValue = static_cast<StringExprAST*>(compStmt->getValue().get());
-            return evaluateCondition(actualValue, expectedValue->getValue(), op);
+            auto *rightValue = static_cast<StringExprAST*>(compStmt->getValue().get());
+            return evaluateCondition(leftValue, rightValue->getValue(), op);
         }
         // This should never get triggered, because invalid type combinations are already filtered out
-        throw std::runtime_error("Internal compiler error - JSON actualValue was string and hardcoded was not");
+        throw std::runtime_error("Internal compiler error - left value was string and right was not");
```
Comment (Splines) [inline compiler/cpp/src/interpreter/interpreter.cpp:123]: Great, this is more specific 👍

File: compiler/cpp/src/lexer/Token.h [hunk 3]
```diff
@@ -43,6 +43,10 @@ enum TokenType {
     TOK_OR, // |
     TOK_EQUALS, // ==
     TOK_NOT_EQUALS, // !=
+    TOK_GREATER_THEN, // >
```
Comment (Splines) [inline compiler/cpp/src/lexer/Token.h:46]: `than` not `then`. Also inconsistent: here you use `TOK_GREATER_THAN` but below without `THAN`: `TOK_GREATER_EQUALS`

File: compiler/cpp/src/lexer/lexer.cpp [hunk 4]
```diff
@@ -61,6 +61,32 @@ Token getTok() {
...
     if (currentContext == PAYLOAD) return consumePayload();
 
+    // Is it a conditional comment identifier?
+    std::string laResult;
+    if (isLookaheadLineCommentChars()) {
+        // Consume LineCommentChars
+        for (int i = 0; i < LineCommentChars.length(); i++) advance();
+        // Update context
+        currentContext = SECTION;
+        return Token(TOK_COM_LINE_IDEN, LineNum, ColNum);
```
Comment (Splines) [inline compiler/cpp/src/lexer/lexer.cpp:71]: `LineNum` and `ColNum` should be lower-caps

File: compiler/cpp/src/lexer/lexer.cpp [hunk 5]
```diff
@@ -61,6 +61,32 @@ Token getTok() {
     // Are we in payload context?
     if (currentContext == PAYLOAD) return consumePayload();
 
+    // Is it a conditional comment identifier?
+    std::string laResult;
```
Comment (Splines) [inline compiler/cpp/src/lexer/lexer.cpp:65]: What does `laResult` stand for?

File: compiler/cpp/src/parser/ast/CompStmtExprAST.h [hunk 6]
```diff
@@ -13,7 +13,11 @@
 
 enum Operator {
     OP_EQUALS,
-    OP_NOT_EQUALS
+    OP_NOT_EQUALS,
+    OP_GREATER_THEN,
```
Comment (Splines) [inline compiler/cpp/src/parser/ast/CompStmtExprAST.h:17]: same here: `than`

File: media/test-file.html [hunk 7]
```diff
@@ -1,9 +1,9 @@
 <html>
     <head>
-        <!--? if var.test == "Title 1" {
+        <!--? if service.frontend[1].label == "Spring Gradle" {
         <title>Title 1</title>
         }-->
-        <!--? if var.test == "Title 2" {
+        <!--? if service.frontend[1].size >= 9 {
```
Comment (Splines) [inline media/test-file.html:6]: nice, love it 👏

**Review by Splines — CHANGES_REQUESTED** [review]

File: compiler/cpp/src/lexer/Token.h (hunk 3 above)
Comment (marcauberer) [reply compiler/cpp/src/lexer/Token.h:46]: Ups. Thanks for making me aware ...
I'd say before we have long token type names such as `TOK_GREATER_EQUALS_THAN`, I will rather rename them like so:
- `TOK_GREATER`
- `TOK_LESS`
- `TOK_GREATER_EQUAL`
- `TOK_LESS_EQUAL`

File: compiler/cpp/src/lexer/lexer.cpp (hunk 4 above)
Comment (marcauberer) [reply compiler/cpp/src/lexer/lexer.cpp:71]: Good catch! Looks like I forgot them when doing the refactoring tasks

File: compiler/cpp/src/lexer/lexer.cpp (hunk 5 above)
Comment (marcauberer) [reply compiler/cpp/src/lexer/lexer.cpp:65]: `laResult` stands for lookahead result

**Comment by marcauberer** [comment]
@Splines everything should be fixed now.

File: compiler/cpp/src/lexer/lexer.cpp (hunk 5 above)
Comment (Splines) [reply compiler/cpp/src/lexer/lexer.cpp:65]: ah ok 😅

File: media/test-data.json [hunk 8]
```diff
@@ -7,13 +7,15 @@
...
                 "type": "frontend",
+                "size": 5,
                 "preselected": false
             },
             {
                 "label": "Spring Gradle",
                 "name": "spring-gradle",
                 "dir": "./spring-gradle",
                 "type": "frontend",
+                "size": 9,
```
Comment (Splines) [inline media/test-data.json:18]: is this change important?

Comment (marcauberer) [reply media/test-data.json:18]: I updated the test file, so that we can test the operators. I would leave it in there

Comment (marcauberer) [reply media/test-data.json:18]: Btw: You also can compare strings. Pretty cool

Comment (Splines) [reply media/test-data.json:18]: ah sorry, I thought that you changed `5` to `9` but I just saw that these were to additions, no line removal

**Review by Splines — APPROVED** [review]

**Pull request merged**